engine/src/main/java/com/redhat/cloud/notifications/events/FromOpenBridgeHistoryFiller.java

```diff
@@ -0,0 +1,137 @@
+    BridgeApiService bridgeApiService;
+
+    @Inject
+    Bridge bridge;
+
+    @Inject
+    BridgeAuth bridgeAuth;
+
+    @Inject
+    NotificationHistoryRepository notificationHistoryRepository;
+
+    @Inject
+    StatelessSessionFactory statelessSessionFactory;
+
+    @Inject
+    MeterRegistry meterRegistry;
+
+    private Counter messagesWithError;
+
+    @CacheName("from-open-bridge-history-filler")
+    Cache seenIds;
+
+    @PostConstruct
+    void init() {
+        messagesWithError = meterRegistry.counter(MESSAGES_WITH_ERROR_NAME);
+    }
+
+    @Scheduled(concurrentExecution = SKIP, every = "${ob.error-check.period:10s}")
+    @Transactional
+    public void execute() {
+
+        if (!featureFlipper.isObBackchannelFillerEnabled()) {
+            return;
+        }
+
+        BridgeItemList<ProcessingError> errorList = bridgeApiService.getProcessingErrors(bridge.getId(), bridgeAuth.getToken());
+
+        // OB offers us a list of items that represent the last X errors
+        // We may have seen individual items before
+        List<ProcessingError> items = errorList.getItems();
+        if (items == null) {
+            return;
+        }
+
+        for (ProcessingError pe : items) {
+
+            // We may have seen an item before, skip it
+            String historyId = pe.getHeaders().get(RHOSE_ORIGINAL_EVENT_ID_HEADER);
+            if (alreadyProcessed(historyId)) {
+                continue;
+            }
+
+            statelessSessionFactory.withSession(statelessSession -> {
+                Map<String, Object> decodedPayload = decodeItem(pe);
+                // TODO reinjectIf needed (?)
+                try {
+                    notificationHistoryRepository.updateHistoryItem(decodedPayload);
+                    messagesWithError.increment();
+                } catch (Exception e) {
+                    Log.warn("|  History update failed", e);
+                }
+            });
+        }
+    }
+
+    private Map<String, Object> decodeItem(ProcessingError pe) {
+        Map<String, Object> map = new HashMap<>();
+        Map<String, String> headers = pe.getHeaders();
+        map.put("historyId", headers.get(RHOSE_ORIGINAL_EVENT_ID_HEADER));
+        map.put("successful", false);
+        Map<String, String> details = new HashMap<>();
+        details.put("originalEvent", pe.getPayload().toString());
+        if (headers.containsKey(DEAD_LETTER_CAUSE)) {
+            String cause = headers.get(DEAD_LETTER_CAUSE);
+            details.put("cause", cause);
+        }
+        map.put("details", details);
+        return map;
+    }
+
+    boolean alreadyProcessed(String historyId) {
+        AtomicBoolean seen = new AtomicBoolean(true);
+        seenIds.get(historyId, k -> {
+            // If the value loader function is invoked, then we've never seen the current historyId.
+            seen.set(false);
+            return DUMMY_CACHE_VALUE;
+        }).await().indefinitely();
+        return seen.get();
+    }
```

Comment:
@pilhuhn FYI I introduced this to cache the seen ids. It relies on a `Cache` from quarkus-cache.